src/Compilers/CSharp/Test/Semantic/Semantics/ExtensionTypeTests.cs

```diff
@@ -5141,18 +5212,19 @@ public void BaseExtension_UnderlyingTypeMismatch_PE()
 """;
 
         var comp = CreateCompilationWithIL(src, ilSource, targetFramework: TargetFramework.Net70);
-        // PROTOTYPE we should check that the extended type is compatible
-        // with the base extension's
         comp.VerifyDiagnostics(
             // (1,27): error CS9216: Extension 'R3' has underlying type 'object' but a base extension has underlying type 'C'.
             // public explicit extension R3 for object : R2 { }
-            Diagnostic(ErrorCode.ERR_UnderlyingTypesMismatch, "R3").WithArguments("R3", "object", "C").WithLocation(1, 27)
+            Diagnostic(ErrorCode.ERR_UnderlyingTypesMismatch, "R3").WithArguments("R3", "object", "C").WithLocation(1, 27),
+            // (2,27): error CS9216: Extension 'R4' has underlying type 'C' but a base extension has underlying type 'object'.
```

Comment:
>// (2,27): error CS9216: Extension 'R4' has underlying type 'C' but a base extension has underlying type 'object'. [](http://example.com/codeflow?start=12&length=114)

The two errors that we report feel very confusing. In fact they contradict each other. Both R3 and R4 have the same declared base extension, but one error sounds like it extends `C` and the other error sounds like it extends `object`. Consider if we should include the name of the problematic base extension into the error. This way the first error would say: "base extension'R2' extends 'C'", and the second error would say: "base extension 'R1' extends 'object'".

Reply:
Good idea. Added the guilty base extension to the diagnostic. Thanks